.github/workflows/testing.yml

```diff
@@ -38,21 +47,49 @@ jobs:
       - name: Setup
         uses: ./.github/actions/setup
 
-      - name: Run integration tests (prod)
+      - name: Setup bun
+        if: matrix.config.bun_version
+        uses: oven-sh/setup-bun@v1
+        with:
+          bun-version: ${{ matrix.config.bun_version }}
+
+      - name: Run integration tests (Prod)
         if: matrix.pinecone_env == 'prod'
         env:
           CI: true
           PINECONE_API_KEY: ${{ secrets.PINECONE_API_KEY }}
-          PINECONE_CONTROLLER_HOST: 'https://api.pinecone.io'
-        run: npm run test:integration:${{ matrix.jest_env }}
+        run: ${{ matrix.config.runner }} run test:integration:${{ matrix.config.jest_env }}
 
-      - name: Run integration tests (staging)
+      - name: Run integration tests (Staging)
         if: matrix.pinecone_env == 'staging'
         env:
           CI: true
-          PINECONE_API_KEY: ${{ secrets.PINECONE_API_KEY_STAGING }}
+          PINECONE_API_KEY: ${{ secrets.PINECONE_API_KEY }}
           PINECONE_CONTROLLER_HOST: 'https://api-staging.pinecone.io'
-        run: npm run test:integration:${{ matrix.jest_env }}
+        run: ${{ matrix.config.runner }} run test:integration:${{ matrix.config.jest_env }}
+
+  integration-test-cleanup:
```

Comment:
We never actually use the integration cleanup, and it does seem like there are cases where we end up with orphaned indexes after the integration suite runs. For now I'm explicitly triggering this after the entire matrix has completed, which should be safe since it just lists indexes and deletes any which are present.